### ISSUE TITLE
Not all console_scripts entry points need pkg_resources from setuptools

### ISSUE THREAD

**Issue by hroncok** [rpm-software-management/rpm#954]
This is a followup after https://github.com/rpm-software-management/rpm/issues/664

Apparently, not all entry_points console_scripts have the pkg_resources import.

It seems to me that packages installed via pip have this:

```
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import re
import sys

from pytest import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

While packages installed via python setup.py install have:

```
#!/usr/bin/python3
# EASY-INSTALL-ENTRY-SCRIPT: 'pytest==...','console_scripts','pytest'
__requires__ = 'pytest==...'
import re
import sys
from pkg_resources import load_entry_point

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
    sys.exit(
        load_entry_point('pytest==...', 'console_scripts', 'pytest')()
    )
```

We could do it the proper way:

 - when console_scripts entrypoints are found, grep them for pkg_resources import and only require setuptools if it is in there

Or the heuristic way:

 - when dist-info, don't add setuptools, when egg-info, do add it

The second way is certainly easier and faster, however I am not yet sure whether the situation is that simple

cc @encukou @ignatenkobrain @Conan-Kudo @gordonmessmer

**Comment by hroncok** [comment]
It seems that dist-info => no pkg_resources, egg-info => pkg_resources is correct.

**Comment by ffesti** [comment]
Should be fixed with #973. Closing.
